### PR TITLE
V2X: splited PR2: add new proto

### PR DESCRIPTION
**Pull request by QIU1995NONAME**
V2X: splited PR2: add forked and new proto

### CODE REVIEW

**Comment by jinghaomiao** [comment]
Please do not check in duplicate changes. 

Also, proto will be designed for backward compatible. So you don't have to worry about that. All modules in Apollo work as a complete system, so re-use existing definitions of proto and gflags is required to minimize the confusion for external developers. You can add new fields in the existing protos to make sure the compatibility is guaranteed.

**Review by jinghaomiao — DISMISSED** [review]
Please do not check in duplicate changes.

Also, proto will be designed for backward compatible. So you don't have to worry about that. All modules in Apollo work as a complete system, so re-use existing definitions of proto and gflags is required to minimize the confusion for external developers. You can add new fields in the existing protos to make sure the compatibility is guaranteed.

File: modules/v2x/proto/v2x_junction.proto [hunk 1]
```diff
@@ -0,0 +1,38 @@
+syntax = "proto2";
+
+package apollo.obu.normal;
```
Comment (storypku) [inline modules/v2x/proto/v2x_junction.proto:3]: package `apollo.v2x.obu` ?

File: modules/v2x/proto/v2x_monitor.proto [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+enum ErrorCode {
+    LTEV = 500;
+    NET = 501;
+    CPU = 502;
+    MEM = 503;
+    GPS = 504;
+    MAP = 510;
+    SPAT = 511;
+    OBUID = 999;
+}
```
Comment (storypku) [inline modules/v2x/proto/v2x_monitor.proto:14]: What does `LTEV` `SPAT`, `OBU` mean?

File: modules/v2x/proto/v2x_obu_rsi.proto [hunk 3]
```diff
@@ -0,0 +1,18 @@
+syntax = "proto2";
+
+package apollo.obu.rsi;
+
+import "modules/common/proto/header.proto";
+import "modules/common/proto/geometry.proto";
+
+message V2x_RSI {
```
Comment (storypku) [inline modules/v2x/proto/v2x_obu_rsi.proto:8]: naming should be uniform.

File: modules/v2x/proto/v2x_obu_traffic_light.proto [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+message NextTrafficLightInfo {
+  enum Color {
+    UNKNOWN = 0;
+    RED = 1;
+    YELLOW = 2;
+    GREEN = 3;
+    BLACK = 4;
+    FLASH_GREEN = 5;
+  };
```
Comment (storypku) [inline modules/v2x/proto/v2x_obu_traffic_light.proto:15]: Color should be defined in a common proto.

File: modules/v2x/proto/v2x_obu_traffic_light.proto [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  optional Color color = 1;
+  optional int32 start_time = 2;
+  optional int32 end_time = 3;
```
Comment (storypku) [inline modules/v2x/proto/v2x_obu_traffic_light.proto:18]: What's the unit of start_time and end_time. Please note it explicitly. And, is init32 large enough for holding time values?

File: modules/v2x/proto/v2x_obu_traffic_light.proto [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+};
+message SingleTrafficLight {
+  enum Color {
+    UNKNOWN = 0;
+    RED = 1;
+    YELLOW = 2;
+    GREEN = 3;
+    BLACK = 4;
+    FLASH_GREEN = 5;
```
Comment (storypku) [inline modules/v2x/proto/v2x_obu_traffic_light.proto:27]: Don't define Color here and there.

File: modules/v2x/proto/v2x_obu_traffic_light.proto [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  optional double gps_x_m = 1;
+  optional double gps_y_m = 2;
+  repeated SingleTrafficLight single_traffic_light = 3;
+}
+
+message RoadTrafficLight {
+  repeated LaneTrafficLight lane_trafficlight = 1;
+  optional int32 road_direction = 2;
+}
+message V2X_IntersectionTrafficLightData {
```
Comment (storypku) [inline modules/v2x/proto/v2x_obu_traffic_light.proto:62]: should the `V2X_` prefix removed. And this file should be renamed to `obu_traffic_light`.

File: modules/v2x/proto/v2x_perception_obstacle.proto [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+
+  optional Point position = 3;
+  optional double theta = 4;
+  optional double length = 5;
+  optional double width = 6;
+  optional double height = 7;
+
+  optional Point velocity = 8;
+
+  enum Type {
```
Comment (storypku) [inline modules/v2x/proto/v2x_perception_obstacle.proto:69]: Wasn't these already defined in Apollo ?
If so, please reuse them.

File: modules/v2x/proto/v2x_rsi.proto [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+import "modules/common/proto/header.proto";
+
+message RSI_Point {
+    optional double x = 1;  // in meters.
+    optional double y = 2;  // in meters.
+}
+message RsiMsg {
+    optional apollo.common.Header header = 1;
+    repeated RSI_Point points = 2;
```
Comment (storypku) [inline modules/v2x/proto/v2x_rsi.proto:13]: `RsiMsg` and `RSI_Point`, should be in uniform.

File: modules/v2x/proto/v2x_traffic_light_detection.proto [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+  repeated TrafficLightBox projected_roi = 10;
+  repeated TrafficLightBox rectified_roi = 11;
+  repeated TrafficLightBox debug_roi = 12;
+}
+message TrafficLight {
+  enum Color {
+    UNKNOWN = 0;
+    RED = 1;
+    YELLOW = 2;
+    GREEN = 3;
```
Comment (storypku) [inline modules/v2x/proto/v2x_traffic_light_detection.proto:34]: Again, color defs should be placed in the **only** common proto file.

**Review by storypku — CHANGES_REQUESTED** [review]

File: modules/v2x/proto/v2x_monitor.proto (hunk 2 above)
Comment (QIU1995NONAME) [reply modules/v2x/proto/v2x_monitor.proto:14]: https://en.wikipedia.org/wiki/Vehicle-to-everything

File: modules/v2x/proto/v2x_junction.proto (hunk 1 above)
Comment (storypku) [inline modules/v2x/proto/v2x_junction.proto:3]: Don't introduce `apollo.obu` namespace.

Comment (storypku) [reply modules/v2x/proto/v2x_junction.proto:3]: Besides, better rename all  `v2x_xxx.proto` files to `xxx.proto`.

**Review by storypku — CHANGES_REQUESTED** [review]

File: modules/v2x/proto/v2x_obu_rsi.proto [hunk 11]
```diff
@@ -0,0 +1,18 @@
+syntax = "proto2";
+
+package apollo.v2x.obu.rsi;
```
Comment (storypku) [inline modules/v2x/proto/v2x_obu_rsi.proto:3]: What about `package apollo.v2x.obu`.

What I mean is, all proto files `v2x_xxx` goes to package `apollo.v2x` and all proto files with name `v2x_obu_` goes to package `apollo.v2x.obu`.

File: modules/v2x/proto/v2x_traffic_light_detection.proto [hunk 12]
```diff
@@ -0,0 +1,58 @@
+syntax = "proto2";
+
+package apollo.v2x.obu.perception;
```
Comment (storypku) [inline modules/v2x/proto/v2x_traffic_light_detection.proto:3]: package apollo.v2x

**Review by storypku — CHANGES_REQUESTED** [review]

**Review by storypku — APPROVED** [review]
Please don't forget to rename proto files with name `v2x_xxx.proto` to `xxx.proto` in later PRs.

**Review by storypku — DISMISSED** [review]

File: modules/v2x/proto/v2x_traffic_light_detection.proto [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+
+  // Traffic light string-ID in the map data.
+  optional string id = 2;
+
+  // How confidence about the detected results, between 0 and 1.
+  optional double confidence = 3 [default = 1.0];
+
+  // Duration of the traffic light since detected.
+  optional double tracking_time = 4;
+
```
Comment (storypku) [inline modules/v2x/proto/v2x_traffic_light_detection.proto:41]: Unit ?

Comment (QIU1995NONAME) [reply modules/v2x/proto/v2x_traffic_light_detection.proto:41]: microsecond

**Review by storypku — APPROVED** [review]

**Pull request merged**